amazon-kinesis-client/src/test/java/software/amazon/kinesis/common/ConfigsBuilderTest.java

```diff
@@ -53,12 +54,27 @@ public class ConfigsBuilderTest {
     private static final String WORKER_IDENTIFIER = "worker-id";
 
     @Test
-    public void testTrackerConstruction() {
+    public void testTrackerConstructionByStreamName() {
         final String streamName = "single-stream";
         final ConfigsBuilder configByName = createConfig(streamName);
         final ConfigsBuilder configBySingleTracker = createConfig(new SingleStreamTracker(streamName));
 
-        for (final ConfigsBuilder cb : Arrays.asList(configByName, configBySingleTracker)) {
+        testTrackerConstruction(streamName, configBySingleTracker, configByName);
+    }
+
+    @Test
+    public void testTrackerConstructionByStreamARN() {
+        final Arn streamARN = Arn.fromString("arn:aws:kinesis:region:123456789012:stream/stream-name");
+        final ConfigsBuilder configByArn = createConfig(streamARN);
+        final ConfigsBuilder configBySingleTracker = createConfig(new SingleStreamTracker(streamARN));
+
+        final String streamName = streamARN.resource().resource();
+        testTrackerConstruction(streamName, configBySingleTracker, configByArn);
+    }
+
+    private void testTrackerConstruction(String streamName,
+                                         ConfigsBuilder expectedConfig, ConfigsBuilder actualConfig) {
```

Comment:
Style: don't do horizontal alignment. This increases blast radius of edits, increases risk of merge conflicts, and provides zero objective benefit.

If your IDE is doing this on its own, a quick workaround is to CRLF before the first param. It should then be properly indented (roughly two indentation levels more than the `private`), and CRLF for every other param will follow that same indentation. Example:
```
private void ...( // observe: first param is on next line
      String streamName,
      ConfigsBuilder ...,
      ConfigsBuilder ...) {
```

---

FYI: you could also KISS and use varargs: `testTrackerConstruction(String streamName, ConfigsBuilder... singleStreamConfig) {`

Recommend refactoring along aspect of single- vs multi-stream, tho. (That seems better to me, and avoids redundancy.)